dhctl/pkg/config/registry/manifests.go

```diff
@@ -0,0 +1,115 @@
+// Copyright 2025 Flant JSC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package registry
+
+import (
+	"fmt"
+	"strings"
+
+	"sigs.k8s.io/yaml"
+
+	"github.com/deckhouse/deckhouse/go_lib/registry/helpers"
+	deckhouse_registry "github.com/deckhouse/deckhouse/go_lib/registry/models/deckhouse-registry"
+)
+
+func newManifestBuilder(modeModel ModeModel, legacyMode bool) *ManifestBuilder {
+	return &ManifestBuilder{
+		modeModel:  modeModel,
+		legacyMode: legacyMode,
+	}
+}
+
+type ManifestBuilder struct {
+	modeModel  ModeModel
+	legacyMode bool
+}
+
+// =======================
+// Secrets
+// =======================
+func (b *ManifestBuilder) DeckhouseRegistrySecretData(getPKI func() (PKI, error)) (map[string][]byte, error) {
```

Comment:
it will be nice to introduce type like type SecretData map[string][]byte and use it as return type